micrometer-core/src/main/java/io/micrometer/core/instrument/Timer.java

```diff
@@ -430,6 +427,18 @@ public Builder description(String description) {
             return super.description(description);
         }
 
+        /**
+         * Convenience method to create new meters from the builder that only differ in
+         * tags. This method can be used for dynamic tagging by creating the builder once
+         * and applying the dynamically changing tags using the returned {@link Function}.
+         * @param registry A registry to add the meter to, if it doesn't already exist.
+         * @return A {@link Function} that returns a meter based on the provided tags.
+         * @since 1.12.0
+         */
+        public Function<Tags, Timer> with(MeterRegistry registry) {
```

Comment:
I think a dedicated type would be useful, since users are going to keep a reference to this to use it. That does increase the importance somewhat on getting the appropriate naming and API before release.